<h1>Activity 1 - PALMER PENGUINS DATA ANALYTICS PROJECT </h1>
<hr>
<h3> Analyst : John Paul R. Nieva </h3>

In [1]:
#import libraries
import numpy as np
import pandas as pd

In [2]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

In [4]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [5]:
penguins.describe(). T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h2>Checking Null Values</h2>

In [6]:
penguins.isnull()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      False   False           False          False              False   
1      False   False           False          False              False   
2      False   False           False          False              False   
3      False   False            True           True               True   
4      False   False           False          False              False   
..       ...     ...             ...            ...                ...   
339    False   False           False          False              False   
340    False   False           False          False              False   
341    False   False           False          False              False   
342    False   False           False          False              False   
343    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 8 columns]

In [7]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h2>Filtering A Column</h2>

In [8]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [9]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [10]:
penguins['species'] == 'Chinstrap'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

In [11]:
penguins[penguins['species'] == 'Chinstrap']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [12]:
adelie_df = penguins[penguins['species'] == 'Chinstrap']
adelie_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [13]:
gentoo_df = penguins[penguins['species'] == 'Gentoo']
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [14]:
penguins['species'] == 'Chinstrap'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

In [15]:
 (penguins['body_mass_g'] >= 4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Name: body_mass_g, Length: 344, dtype: bool

In [16]:
(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Length: 344, dtype: bool

In [17]:
penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

In [18]:
chinstrap_heavy = penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)]
chinstrap_heavy

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

<h2> Filling UP Null Values <h2>

In [19]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [20]:
penguins['bill_length_mm'] 

0      39.1
1      39.5
2      40.3
3       NaN
4      36.7
       ... 
339    55.8
340    43.5
341    49.6
342    50.8
343    50.2
Name: bill_length_mm, Length: 344, dtype: float64

In [21]:
penguins['bill_length_mm'] . mean()

43.9219298245614

In [22]:
penguins['bill_length_mm'] = penguins['bill_length_mm'] .fillna(penguins['bill_length_mm'].mean())

In [23]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'] .fillna(penguins['bill_depth_mm'].mean())


In [24]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'] .fillna(penguins['flipper_length_mm'].mean())


In [25]:
penguins['body_mass_g'] = penguins['body_mass_g'] .fillna(penguins['body_mass_g'].mean())


In [26]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [27]:
penguins['sex'] = penguins['sex'] .fillna(penguins['sex'] . mode()[0])

In [28]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h2> CHANGING THE DATA TYPE <h2>

In [29]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')
penguins['year'] = penguins['year'].astype('category')





In [30]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


In [31]:
penguins['body_mass_g'].mean()

4201.754385964912

In [32]:
penguins['species'] == 'Gentoo'

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Name: species, Length: 344, dtype: bool

<h4>How to Create a Column</h4>

In [33]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


In [34]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [35]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000 , 1)

In [36]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

In [37]:
penguins['mass_kg'].max()

6.3

penguins[penguins['mass_kg'] == 6.3]

<h2> EXPLANATORY DATA ANALYTICS <h2>

<h2>QUESTION 1</h2>
<h4>What is the average bill depth of gentoo in the year 2008?</h4>

In [38]:
gentoo_2008 = penguins[(penguins["species"] == "Gentoo") & (penguins["year"] == 2008)]
gentoo_2008

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
186  Gentoo  Biscoe            49.1           14.8              220.0   
187  Gentoo  Biscoe            48.4           16.3              220.0   
188  Gentoo  Biscoe            42.6           13.7              213.0   
189  Gentoo  Biscoe            44.4           17.3              219.0   
190  Gentoo  Biscoe            44.0           13.6              208.0   
191  Gentoo  Biscoe            48.7           15.7              208.0   
192  Gentoo  Biscoe            42.7           13.7              208.0   
193  Gentoo  Biscoe            49.6           16.0              225.0   
194  Gentoo  Biscoe            45.3           13.7              210.0   
195  Gentoo  Biscoe            49.6           15.0              216.0   
196  Gentoo  Biscoe            50.5           15.9              222.0   
197  Gentoo  Biscoe            43.6           13.9              217.0   
198  Gentoo  Biscoe            45.5           13.9              210.0   
199  Gentoo  Biscoe            50.5           15.9              225.0   
200  Gentoo  Biscoe            44.9           13.3              213.0   
201  Gentoo  Biscoe            45.2           15.8              215.0   
202  Gentoo  Biscoe            46.6           14.2              210.0   
203  Gentoo  Biscoe            48.5           14.1              220.0   
204  Gentoo  Biscoe            45.1           14.4              210.0   
205  Gentoo  Biscoe            50.1           15.0              225.0   
206  Gentoo  Biscoe            46.5           14.4              217.0   
207  Gentoo  Biscoe            45.0           15.4              220.0   
208  Gentoo  Biscoe            43.8           13.9              208.0   
209  Gentoo  Biscoe            45.5           15.0              220.0   
210  Gentoo  Biscoe            43.2           14.5              208.0   
211  Gentoo  Biscoe            50.4           15.3              224.0   
212  Gentoo  Biscoe            45.3           13.8              208.0   
213  Gentoo  Biscoe            46.2           14.9              221.0   
214  Gentoo  Biscoe            45.7           13.9              214.0   
215  Gentoo  Biscoe            54.3           15.7              231.0   
216  Gentoo  Biscoe            45.8           14.2              219.0   
217  Gentoo  Biscoe            49.8           16.8              230.0   
218  Gentoo  Biscoe            46.2           14.4              214.0   
219  Gentoo  Biscoe            49.5           16.2              229.0   
220  Gentoo  Biscoe            43.5           14.2              220.0   
221  Gentoo  Biscoe            50.7           15.0              223.0   
222  Gentoo  Biscoe            47.7           15.0              216.0   
223  Gentoo  Biscoe            46.4           15.6              221.0   
224  Gentoo  Biscoe            48.2           15.6              221.0   
225  Gentoo  Biscoe            46.5           14.8              217.0   
226  Gentoo  Biscoe            46.4           15.0              216.0   
227  Gentoo  Biscoe            48.6           16.0              230.0   
228  Gentoo  Biscoe            47.5           14.2              209.0   
229  Gentoo  Biscoe            51.1           16.3              220.0   
230  Gentoo  Biscoe            45.2           13.8              215.0   
231  Gentoo  Biscoe            45.2           16.4              223.0   

     body_mass_g     sex  year  mass_kg  
186       5150.0  female  2008      5.2  
187       5400.0    male  2008      5.4  
188       4950.0  female  2008      5.0  
189       5250.0    male  2008      5.2  
190       4350.0  female  2008      4.4  
191       5350.0    male  2008      5.4  
192       3950.0  female  2008      4.0  
193       5700.0    male  2008      5.7  
194       4300.0  female  2008      4.3  
195       4750.0    male  2008      4.8  
196       5550.0    male  2008      5.6  
197       4900.0  female  2008      4.9  
198       4200.0  female  

In [39]:
average_bill_depth = gentoo_2008["bill_depth_mm"].mean()
average_bill_depth

14.923913043478258

<h2>INSIGHT NO.1</h2>
<h4>The average bill depth of gentoo in the year 2008 is 14.9</h4>

<h2>QUESTION 2</h2>
<h4>From which sex of Chinstrap has the longest bill depth?</h4>

In [40]:
chinstrap = penguins[penguins["species"] == "Chinstrap"]
longest_bill = chinstrap[chinstrap["bill_depth_mm"] == chinstrap["bill_depth_mm"].max()]
longest_bill

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
307  Chinstrap  Dream            54.2           20.8              201.0   

     body_mass_g   sex  year  mass_kg  
307       4300.0  male  2008      4.3

<h2>INSIGHT.2</h2>
<h4>Insight: This shows that the male chinstrap has the longest bill depth having 20.8 depth.</h4>

<h2>QUESTION 3</h2>

<h4>What is the average flipper length of Gentoo penguins on Biscoe island?</h4>

In [41]:
penguins[(penguins["species"] == "Gentoo") & (penguins["island"] == "Biscoe")]["flipper_length_mm"].mean()

217.05576777966422

<h2>INSIGHT.3</h2>
<h4>Average flipper length for Gentoo penguins on Biscoe island is 217mm</h4>

<h2>QUESTION 4</h2>
<h4>What is the average body mass of Chinstrap penguins found on Dream island?</h4>

In [42]:
penguins[(penguins["species"] == "Chinstrap") & (penguins["island"] == "Dream")]["body_mass_g"].mean()

3733.0882352941176

<h2>INSIGHT.4</h2>
<h4>Average weight for Chinstrap penguins on Dream island is 3733g</h4>

<h2>QUESTION 5</h2>
<h4>Which species has the smallest bill depth?</h4>

In [43]:
species_smallest_bill_depth = penguins[penguins["bill_depth_mm"] == penguins["bill_depth_mm"].min()]["species"].values[0]
species_smallest_bill_depth

'Gentoo'

<h2>INSIGHT.5</h2>
<h4>Gentoo has smallest bill depth among all the species</h4>

<h2>QUESTION 6</h2>
<h4>Which penguin is the longest in terms of flipper length?</h4>

In [44]:
longest_flipper_row = penguins[penguins["flipper_length_mm"] == penguins["flipper_length_mm"].max()]
species_longest_flipper = longest_flipper_row["species"].values[0]
longest_flipper_row



species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
215  Gentoo  Biscoe            54.3           15.7              231.0   

     body_mass_g   sex  year  mass_kg  
215       5650.0  male  2008      5.6

In [45]:
species_longest_flipper

'Gentoo'

<h2>INSIGHT.6</h2>
<h4>Gentoo has a longest flipper length having 231mm </h4>

<h2>QUESTION 7</h2>
<h4>What is the bill length (mm) of a Chinstrap penguin found on Dream island?</h4>

In [46]:
chinstrap_bl = penguins[(penguins["species"] == "Chinstrap") & (penguins["island"] == "Dream")]["bill_length_mm"]
print(chinstrap_bl)


276    46.5
277    50.0
278    51.3
279    45.4
280    52.7
       ... 
339    55.8
340    43.5
341    49.6
342    50.8
343    50.2
Name: bill_length_mm, Length: 68, dtype: float64


<H2>INSIGHT.7</H2>
<h4>This show the bill length of chinstrap penguin found on dream island</h4>

<h2>QUESTION 8</h2>
<h1>What is the bill length of the first Chinstrap penguin on Dream Island?</h1>

In [60]:
chinstrap_dream = penguins[(penguins["species"] == "Chinstrap") & (penguins["island"] == "Dream")]
first_bill = chinstrap_dream["bill_length_mm"].values[0]
print("Bill length of the first Chinstrap penguin:", first_bill)


Bill length of the first Chinstrap penguin: 46.5


<h2>INSIGHT.8</h2>
<H4>Using .values[0] gives the first value from the column as a basic number (not as a pandas Series). It’s useful if you're just starting out and want a simple result.</H4>

<H2>QUESTION 9</H2>
<H2>What is the range (min and max) of bill lengths for Chinstrap penguins on Dream Island?</H2>

In [49]:
min_bl = chinstrap_bl.min()
max_bl = chinstrap_bl.max()
print(f"Min bill length: {min_bl}, Max bill length: {max_bl}")


Min bill length: 40.9, Max bill length: 58.0


<h2>INSIGHT.9</h2>
<h4>The minimum and maximum of bill lenghts of a chinstrap penguin from Dream Island are : Min:40.9 Max:58.0</h4>

<H2>QUESTION 10</H2>
<H4>Is the average bill length bigger for heavier penguins?</H4>

In [58]:
avg_bill_heavy = heavier["bill_length_mm"].mean()
avg_bill_light = lighter["bill_length_mm"].mean()

print("Average bill length (heavier):", avg_bill_heavy)
print("Average bill length (lighter):", avg_bill_light)



Average bill length (heavier): 50.38387096774194
Average bill length (lighter): 47.535135135135135


<h2>INSIGHT.10</h2>
<h4>See if heavier penguins tend to have longer bills.</h4>

<H2>QUESTION 11</H2>
<H4>Is the average flipper length different for heavier vs lighter penguins? (Split by body mass above/below average)</H4>

In [56]:
average_mass = chinstrap_dream["body_mass_g"].mean()
heavier = chinstrap_dream[chinstrap_dream["body_mass_g"] > average_mass]
lighter = chinstrap_dream[chinstrap_dream["body_mass_g"] <= average_mass]

print("Average flipper length of heavier penguins:", heavier["flipper_length_mm"].mean())
print("Average flipper length of lighter penguins:", lighter["flipper_length_mm"].mean())


Average flipper length of heavier penguins: 200.7741935483871
Average flipper length of lighter penguins: 191.67567567567568


<H2>INSIGHT.11</H2>
<H4>See if heavier penguins tend to have longer or shorter flippers than lighter ones.</H4>

<H2>QUESTION 12</H2>
<H4>Is the average body mass higher when the average flipper length is longer?</H4>

In [54]:
avg_body_mass = chinstrap_dream["body_mass_g"].mean()
avg_flipper_length = chinstrap_dream["flipper_length_mm"].mean()
print("Average body mass:", avg_body_mass)
print("Average flipper length:", avg_flipper_length)


Average body mass: 3733.0882352941176
Average flipper length: 195.8235294117647


<H2>INSGHT.12</H2>
<H4>Compare these two averages to see if heavier penguins tend to have longer flippers.</H4>

<H2>QUESTION 13</H2>
<H4>Does a bigger average bill length come with a bigger average bill depth?</H4>

In [55]:
avg_bill_length = chinstrap_dream["bill_length_mm"].mean()
avg_bill_depth = chinstrap_dream["bill_depth_mm"].mean()
print("Average bill length:", avg_bill_length)
print("Average bill depth:", avg_bill_depth)


Average bill length: 48.83382352941177
Average bill depth: 18.42058823529412


<H2>INSIGHT.13</H2>
<H4>The average bill length (48.8 mm) is considerably larger than the average bill depth (18.4 mm), but that does not imply a direct relationship between them.</H4>